src/addrman.h

```diff
@@ -410,9 +413,20 @@ friend class CAddrManTest;
 
         OverrideStream<Stream> s(&s_, s_.GetType(), stream_version);
 
-        unsigned char nKeySize;
-        s >> nKeySize;
-        if (nKeySize != 32) throw std::ios_base::failure("Incorrect keysize in addrman deserialization");
+        //! The maximum format this software can Unserialize().
+        static constexpr Format maximum_supported_format = Format::V3_BIP155;
+
+        uint8_t compat;
+        s >> compat;
+        const uint8_t lowest_compatible = compat - m_incompatibility_base;
+        if (lowest_compatible > maximum_supported_format) {
+            throw std::ios_base::failure(
+                strprintf("Unsupported format of addrman database: %u. It is compatible with "
+                          "formats >=%u, but the maximum supported by this version of %s is %u. "
+                          "Continuing operation without using the saved list of peers.",
```

Comment:
stye-nit: I think the "continue" (and logging) is responsibility (and is already done) by the calling code that also catches the exception. No need to assume this exception is always caught.

Reply:
I agree. Removed the last sentence. Alas, the code that catches this exception is too generic and we can't print "Continuing operation without using the saved list of peers" from it.